src/EditorFeatures/CSharpTest/UseCompoundAssignment/UseCompoundAssignmentTests.cs

```diff
@@ -0,0 +1,561 @@
+        public async Task TestMultiAccess()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    public int a;
+}
+
+public class D
+{
+    C c;
+
+    void M()
+    {
+        this.c.a [||]= this.c.a + 10;
+    }
+}",
+@"public class C
+{
+    public int a;
+}
+
+public class D
+{
+    C c;
+
+    void M()
+    {
+        this.c.a += 10;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestNotOnProp()
+        {
+            await TestMissingAsync(
+@"public class C
+{
+    int a { get; set; }
+
+    void M()
+    {
+        a [||]= a + 10;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestNotOnUnboundSymbol()
+        {
+            await TestMissingAsync(
+@"public class C
+{
+    void M()
+    {
+        a [||]= a + 10;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestNotWithSideEffects()
+        {
+            await TestMissingAsync(
+@"public class C
+{
+    int i;
+
+    C Goo() => this;
+
+    void M()
+    {
+        this.Goo().i [||]= this.Goo().i + 10;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestTrivia()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    void M(int a)
+    {
+        // before
+        a [||]= a + 10; // after
+    }
+}",
+@"public class C
+{
+    void M(int a)
+    {
+        // before
+        a += 10; // after
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestTrivia2()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    void M(int a)
+    {
+        a /*mid1*/ [||]= /*mid2*/ a + 10;
+    }
+}",
+@"public class C
+{
+    void M(int a)
+    {
+        a /*mid1*/ += /*mid2*/ 10;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestFixAll()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    void M(int a, int b)
+    {
+        a {|FixAllInDocument:|}= a + 10;
+        b = b - a;
+    }
+}",
+@"public class C
+{
+    void M(int a, int b)
+    {
+        a += 10;
+        b -= a;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestNestedAssignment()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    void M(int a, int b)
+    {
+        b = (a [||]= a + 10);
+    }
+}",
+@"public class C
+{
+    void M(int a, int b)
+    {
+        b = (a += 10);
+    }
+}");
+        }
+    }
+}
```

Comment:
I'd want to include one test for `a = a + 10;` where `a` was not defined.
Similarly, `this.a = this.a + 10` with undefined `a`.